### ISSUE TITLE
Inconsistent Inspector panel behavior when clicking a node vs. a file

### ISSUE THREAD

**Issue by BigZaphod** [godotengine/godot#74620]
### Godot version

v4.0.stable.official [92bee43ad]

### System information

macOS 13.2.1 (22D68), 16 inch MacBook Pro M1 Max

### Issue description

I am constantly getting momentarily confused about what's appearing in the Inspector tab. If I single click a node in the scene tree, that node becomes selected and the information for that node is in the Inspector tab as I would expect. HOWEVER if I single click a file in the FileSystem tab, all that happens is that the file becomes visually selected - it does not change the content of the Inspector tab despite my having just clicked it and despite there being a blue outline around the entire FileSystem tab's inner content indicating that focus is there.

### Steps to reproduce

This screenshot isn't terribly exciting, but it shows what happens *after* I clicked on the "icon.svg" file in the FileSystem tab. Note that the Inspector is still showing the information for a node I had selected previously. Also note that a quick glance at this window makes it appear as if two different objects are selected at the same time - the CanvasGroup node in the scene, and also the icon.svg file.

<img width="1338" alt="Screenshot 2023-03-08 at 2 59 56 PM" src="https://user-images.githubusercontent.com/453721/223849773-78a135bc-0d05-472e-833d-1e477a719d96.png">

This is very confusing, IMO.

There are multiple problems here, I think:

One is that single clicking on a file in the FileSystem tab only selects it in that tab but doesn't open the information for that file in the Inspector. I instead have to double click file objects to show them in the Inspector tab - unlike everything else which is a single click (as far as I know).

The other problem is that multiple things can look selected at the same time even when they are not. In the screenshot above, the CanvasGroup node shouldn't look selected because I had just clicked on the icon.svg file and that is where my focus is. You could argue that this is fine because I didn't "fully select" icon.png by double clicking it - but indeed even when I do double click icon.svg to "open" the file, now it shows up only in the inspector but nothing else has changed and no new editor has appeared - it still looks like CanvasGroup is selected as well except now the Inspector properties don't make sense.

<img width="1338" alt="Screenshot 2023-03-08 at 3 13 14 PM" src="https://user-images.githubusercontent.com/453721/223851351-1f14a7d3-81c9-415a-ad3c-9753665b7c53.png">

I realize that there's probably some complication here because double clicking a file is how you "open" it and I don't think I would want a single click of a scene file to immediately open it up in a new tab (although... maybe that wouldn't be terrible?), but this is fuzzy when it comes to other types of files like images and resources which appear to "open" in the inspector.

I don't entirely know what the solution is, just that the current situation can be confusing and disorienting.

### Minimal reproduction project

Any project can reproduce this.

**Comment by AThousandShips** [comment]
It's to prevent loading things by accident when selecting them, otherwise it'd work different from when clicking a scene in the file systen

**Comment by BigZaphod** [comment]
Yes, I mentioned that myself - I assumed this is why. That doesn't change the fact that this is extremely confusing and some aspect of the behavior should be fixed, IMO. Even just clearing prior selections and clearing the Inspector when I single click on a file would be an improvement. That would at least remind me to double click to "open" that file if I want to view it in the inspector. Clicking on a different file should once again reset everything. There shouldn't be multiple things on screen marked in blue "selected" state that don't match what the Inspector is showing.

**Comment by AThousandShips** [comment]
Also clearing the inspector when selecting a file would remove an important feature, you couldn't drag and drop files into it any longer

**Comment by AThousandShips** [comment]
See [here](https://github.com/godotengine/godot-proposals/discussions/6339#discussioncomment-5186885) for an explanation

**Comment by BigZaphod** [comment]
I'm not sure why you're being so dismissive? There are solutions to this. This isn't an impossible to solve problem. Fire up Unity and select a scene object there and then select a different one in the project navigator - boom it deselects your scene object and immediately shows you what you clicked on in the inspector. This is logical and consistent and there's only one "thing" selected at a time.

The problem of drag and dropping files is also totally solvable - and indeed somehow magically works in Unity, too! When I'm dragging, I still have the mouse button down. It doesn't need to change the selection at that instant! Wait until I lift the mouse button. In virtually all other software, mouse down is only a "highlighting" effect and the selection doesn't change until mouse up. Godot is the one doing the odd thing here.

**Comment by AThousandShips** [comment]
This doesn't need a solution, this is by design, please don't come into the bug report dead set on that you're right, it's not helpful

**Comment by YuriSizov** [comment]
Inspector does not show currently selected objects, it shows currently edited objects. Scene by itself is not displayed as an object in the inspector, but its nodes are. So as you edit scene, selecting nodes makes them active as edited objects. Selecting a file does not open it for editing, but opening it does make it an edited object, and thus makes it appear in the inspector.

In short, you should not look at it in terms of a focus, but in terms of an editing context. Making files editable with a single click is not impossible to implement, though. You should open an [enhancement proposal](https://github.com/godotengine/godot-proposals), explaining why that would be an improvement to your workflow.

**Comment by BigZaphod** [comment]
> This doesn't need a solution, this is by design, please don't come into the bug report dead set on that you're right, it's not helpful

This is extremely arrogant. Bad usability is a bug. Does this project want feedback or not?

**Comment by YuriSizov** [comment]
I think what @AThousandShips wants to say is that while it may not be what you intuitively expect to happen, there is a reason for this behavior, and it works and makes sense to other users.

That said, @AThousandShips, you did put it in a rather dismissive way. Please be mindful that different users may find different workflows frustrating, even if it makes sense to you.

**Comment by BigZaphod** [comment]
If the correct thing to do is to file this as an enhancement request, then sure, I can do that, but personally this feels more like a bug - I select something and the prior selection is not being unselected so now I have two things selected. I thought something like an enhancement proposal was for huge features or something. I don't know who is supposed to decide which things are bugs vs. enhancements - especially when it comes to usability type of stuff which will always be somewhat subjective.

**Comment by YuriSizov** [comment]
@BigZaphod Well, I am here to help you with that decision :) And so are other people with the Member badge. Everything worth improving can be argued to be considered a bug, but the important difference here is that the current behavior is intended and is working as designed. Also, usability is not an exact science and it doesn't always happen to have solutions that apply universally. There is a reason why Godot clicks with people where others engines don't, and there is a reason why you clicked better with Unity's behavior in this case - everyone thinks differently, and tools that work for some may not work for others.

As such, usability improvements are usually about making sure that we don't make people who are used to or prefer Godot's ways uncomfortable, while allowing customization options or alternative tools for those who need them. While you may feel strongly about this issue, we can't consider it a bug based only on that. And a proposal is the best place to discuss that, and to turn your issue into something actionable.

**Comment by BigZaphod** [comment]
@YuriSizov ok I'll post it as a proposal, I guess, but this isn't something I'm going to go to war over or even care that deeply about. I only use Godot as an intermittent hobby and it was just a suggestion and a data point that I was hoping could be used to make things better someday.

I'm honestly not in a good mood right now having been jumped all over for this after taking 30 minutes to write it up in the first place and now still discussing the meta 30 minutes later. It honestly puts me off the idea of submitting issues in the future. I was also going to raise one about the difficulty of navigating the 3D editor using just a MacBook trackpad, but this experience makes me think that's not worth reporting that either because someone will probably say it wasn't designed for that use case so I am not justified even bringing it up or something like that. 😕

**Comment by YuriSizov** [comment]
You are justified to bring up any issue that you face. We use this repository only for bug reports, and we use a separate repository for enhancements, because we have very many of both and need some organization. Making a proposal issue or opening a proposal discussion is just as valid to bring an issue to the attention of Godot users and contributors.

But you have to keep in mind that others may not share your concerns and may be happy with things as they are. That's always a possibility, especially with UX.

It's not a big deal that you picked a wrong repository to open your issue. GitHub doesn't make it easy for us to manage such a large project, so we can't very effectively guide everyone to the right place and save you time. But that's why I am here talking to you, so you know what to do in the future.
